### PR TITLE
Clarifications about ports in 0.17

### PR DESCRIPTION
**Pull request by kitofr**
Add section about `port modules` and explain how to add one to a project, and what happens if you don't.
Try to clarify how to upgrade outgoing ports and where to call them.

### CODE REVIEW

File: upgrade-docs/0.17.md [hunk 1]
```diff
@@ -208,6 +208,32 @@ The type of `main` has changed from `Signal Html` to `Program flags`. The main v
...
+30| port focus : String -> Cmd msg
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All ports must be defined in a `port module`. You should probably have just one
+of these for your project. This way all of your foreign interactions stay
+relatively organized.
+
+Detected errors in 1 module.
+```
+
+### Incomming / Outgoing ports
```
Comment (eliasdorneles) [inline upgrade-docs/0.17.md]: `Incomming` <- small typo

(disclaimer: I'm not a committer, just an user passing by. +1 to improving upgrade docs anyway :) )

**Comment by eeue56** [comment]
@evancz quite a few people getting confused by this (seems like the compiler hint doesn't help enough, this would be handy)

**Comment by despairblue** [comment]
Thanks, that helped a bit. Still trying to figure out how to import that module (`I cannot find module 'Ports'`) and after that how to use an incoming port to initialize the app.

**Comment by despairblue** [comment]
Oh, by looking at [elm-todomvc](https://github.com/evancz/elm-todomvc/blob/master/Todo.elm#L1) I figured out what was expected of me :smile: 

I really thought I have to make a module called `Ports` and import it.

File: upgrade-docs/0.17.md [hunk 2]
```diff
@@ -208,6 +208,32 @@ The type of `main` has changed from `Signal Html` to `Program flags`. The main v
 
 Talking to JavaScript still uses ports. It is pretty similar, but adapted to fit nicely with commands and subscriptions.
 
+### Port modules
+For house keeping sake there's also a new concept called `port module`. All ports must be defined inside a `port module` and the recommendation is to stick to one per project.
```
Comment (DestyNova) [inline upgrade-docs/0.17.md]: ^ "For housekeeping's sake"

**Comment by evancz** [comment]
I think it's too late. Sorry I did not update this sooner. @eeue56, can you reach out to me in other channels if you think there's something serious. Enough random people ping my @ that it is too much noise to respond to reliably.

**Pull request closed without merging**